simpleclient_httpserver/src/main/java/io/prometheus/client/exporter/BasicAuthenticator.java

```diff
@@ -0,0 +1,88 @@
+package io.prometheus.client.exporter;
+
+/**
+ * Basic HTTP authenticator
+ */
+public class BasicAuthenticator extends com.sun.net.httpserver.BasicAuthenticator {
+
+    private static final String REALM = "/";
+
+    private String username;
+    private String password;
+
+    private BasicAuthenticator(String username, String password) {
+        super(REALM);
+        this.username = username;
+        this.password = password;
+    }
+
+    /**
+     * Check username and password
+     * @param username
+     * @param password
+     * @return
+     */
+    @Override
+    public boolean checkCredentials(String username, String password) {
+        if ((this.username == null) || (this.username.trim().length() == 0)) {
+            return false;
+        }
+
+        if ((this.password == null) || (this.password.trim().length() == 0)) {
+            return false;
+        }
```

Comment:
`this.username` and `this.password` cannot be `null` or empty, so there's no need to check this here.

Reply:
Agreed.